source/ProcessManager.Orchestrations/InternalProcesses/MigrateCalculationsFromWholesale/V1/Activities/MigrateCalculationActivity_MigrateCalculationsFromWholesale_V1.cs

```
@@ -38,10 +53,129 @@ public static Guid GetMigratedWholesaleCalculationIdCustomStateGuid(Orchestratio
     public async Task<string> Run(
         [ActivityTrigger] ActivityInput input)
     {
-        // TODO: Implement migration
-        await Task.CompletedTask.ConfigureAwait(false);
+        try
+        {
+            var wasMigrated = await WasCalculationIdMigratedAsync(input.CalculationToMigrateId).ConfigureAwait(false);
+            if (wasMigrated)
+                return $"Migration skipped for '{input.CalculationToMigrateId}'.";
```

This should not be possible to reach? It should throw an exception instead of just skipping the migration, since this would mean a mistake in retreiving the calculations to migrate.

---

Sure, but wouldn't it have been better that it threw an exception so you would have noticed immediately what was wrong in the test output? 😅

---

No. This activity doesn't know anything about what the code before it did - it just have to be idempotent. :)